client/src/components/Ticketing/ticketingmanager/showings/ShowingUpdated/EventShowingContainer.tsx

```diff
@@ -0,0 +1,64 @@
+import {Showing} from '../../../../../interfaces/showing.interface';
+import React, {useState} from 'react';
+import {EventShowingForm} from './EventShowingForm';
+import {EventShowingView} from './EventShowingView';
+import {useEvent} from './EventProvider';
+import {createDeleteFunction, createSubmitFunction} from './ShowingUtils';
+
+interface EventShowingContainerProps {
+  showing: Showing;
+}
+
+export const EventShowingContainer = (props: EventShowingContainerProps) => {
+  const {showing} = props;
+  const {setReloadShowing, token, setEditing, setPopUpProps} = useEvent();
+  const [edit, setEdit] = useState(false);
+
+  const onSuccess = async (event) => {
+    setReloadShowing((reload) => !reload);
+    setEdit((edit) => !edit);
+    setEditing((editing) => !editing);
+    setPopUpProps('Success', 'Showing successfully updated', true);
+  };
+
+  const onError = async (event) => {
+    try {
+      const res = await event.json();
+      setPopUpProps('Failure', res.error, false);
+    } catch (error) {
+      setPopUpProps('Failure', 'Showing update failed', false);
+    }
+  };
+
+  const onEditClick = async () => {
+    setEditing((edit) => !edit);
+    setEdit((edit) => !edit);
+  };
```

Comment:
React state update are automatically batched and updated asynchronously so `async` keyword not required. Just a note, I don't think it really affects anything if removed or not.

Reply:
Updated